actors/badge/src/main/java/org/sunbird/badge/actors/BadgeAssertionActor.java

```diff
@@ -61,21 +62,28 @@ public void onReceive(Request request) throws Throwable {
    */
   private void createAssertion(Request actorMessage) throws IOException {
     ProjectLogger.log(
-        "Got request to create badge assertion", actorMessage.getRequest(), LoggerEnum.INFO.name());
+        "BadgeAssertionActor:createAssertion: Got request to create badge assertion",
+        actorMessage.getRequest(),
+        LoggerEnum.INFO.name());
+    String recipientId = (String) actorMessage.getRequest().get(BadgingJsonKey.RECIPIENT_ID);
+    String objectType = (String) actorMessage.getRequest().get(BadgingJsonKey.RECIPIENT_TYPE);
+    String badgeId = (String) actorMessage.getRequest().get(BadgingJsonKey.BADGE_ID);
+    BadgeAssertionValidator.checkBadgeAndRecipientFromSameRootOrg(recipientId, objectType, badgeId);
```

Comment:
change  checkBadgeAndRecipientFromSameRootOrg to validateRootOrg 
this method will internally call validateUserRootOrg  or validateContentRootOrg based on  objectType.